c10/util/ThreadLocal.h

```diff
@@ -141,6 +141,14 @@ class ThreadLocal {
     return &var;                              \
   })
 
+#define C10_DEFINE_TLS(Type, Name)     \
+  ::c10::ThreadLocal<Type> Name([]() { \
+    thread_local Type var;             \
+    return &var;                       \
+  })
+
+#define C10_DECLARE_TLS_extern(Type, Name) extern ::c10::ThreadLocal<Type> Name
+
```

Comment:
We probably won't need these additional definitions if we move the thread local definition out of the header (see below)

Reply:
removed